src/ua_parser/basic.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+import io
+import os
+from itertools import starmap
+from operator import methodcaller
+from typing import List
+
+from .core import (
+    Parser as AbstractParser,
+    PartialParseResult,
+    Domain,
+    UserAgent,
+    OS,
+    Device,
+    Matchers,
+    UserAgentMatcher,
+    OSMatcher,
+    DeviceMatcher,
+)
+
+
+class Parser(AbstractParser):
+    """A simple pure-python parser based around trying a numer of regular
+    expressions in sequence for each domain, and returning a result
+    when one matches.
+    """
+
+    user_agent_parsers: List[UserAgentMatcher]
+    os_parsers: List[OSMatcher]
+    device_parsers: List[DeviceMatcher]
+
+    def __init__(
+        self,
+        matchers: Matchers,
+    ) -> None:
+        self.user_agent_parsers = matchers[0]
+        self.os_parsers = matchers[1]
+        self.device_parsers = matchers[2]
```

Comment:
```suggestion
        self.user_agent_matchers = matchers[0]
        self.os_matchers = matchers[1]
        self.device_matchers = matchers[2]
```
there are probably others